posydon/popsyn/binarypopulation.py

```diff
@@ -105,10 +107,20 @@ def __init__(self, **kwargs):
         entropy = self.kwargs.get('entropy', None)
         seq = np.random.SeedSequence(entropy=entropy)
 
```

Comment:
I am realizing now that removal of MPI in this fashion will also remove functionality for users to parallelize populations locally without a compute cluster. Unless we want to implement multiprocessing as well, MPI was an easy way for the code to be parallelized locally & I remember this being an important part of code marketability, even if in practice we would use slurm 90% of the time.

Is it not possible to keep some of the old MPI communicator code and when running with job arrays and setting `comm=None`? Or is it the case that we are removing MPI permanently from the code base?

Reply:
For running it on a single computer, what do we need from MPI, what we can't do with splitting the process and let things run in child processes with one parent keep track of all?

Reply:
This is true. This should totally be fine. I can add those components back in. Since this version doesn't use the `use_MPI` flag, it should be possible to have both exist side-by-side without requiring this part of the code to initialise MPI.

Reply:
> For running it on a single computer, what do we need from MPI, what we can't do with splitting the process and let things run in child processes with one parent keep track of all?

Could you provide a bit more information on this?
To me it seems that if `mpi4py` is not imported, except if used locally, it shouldn't interfere with SLURM's MPI.

Reply:
Yes, that's the point we can't import it. But usually in python you try to import stuff at the beginning to get script checks enabled for the imported parts. If we import stuff only under a condition during running the script, the syntax check for the imported script does happen at this moment and not at the beginning of running the script. It should be fine for an external module, but we need to be careful with such things.

Reply:
I've put back the check to import MPI. It only imports if `use_MPI` is set to True. Otherwise it is not imported. Moreover, additional checks are put in place to make sure that `use_MPI` and job arrays are not used at the same time.